var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -366,41 +366,42 @@ def install(self, spec, prefix):
         else:
             options.append('--with-clanguage=C')
 
-        # PETSc depends on scalapack when '+mumps' (see depends())
-        # help PETSc pick up Scalapack from MKL
-        if spec.satisfies('+mumps'):
-            scalapack = spec['scalapack'].libs
-            options.extend([
-                '--with-scalapack-lib=%s' % scalapack.joined(),
-                '--with-scalapack=1'
-            ])
-        else:
-            options.extend([
-                '--with-scalapack=0'
-            ])
-
         # Activates library support if needed (i.e. direct dependency)
-        if '^libjpeg-turbo' in spec:
-            jpeg_library = ('libjpeg-turbo', 'libjpeg')
-        else:
-            jpeg_library = 'libjpeg'
+        for libjpeg_spackpkg in ('libjpeg-turbo', 'libjpeg'):
+            if libjpeg_spackpkg in spec.dependencies_dict():
+                break
+
+        for scalapack_spackpkg in ('netlib-scalapack', 'intel-mkl', 'intel-oneapi'):
```

Comment:
@tgamblin is there a way to query spack to get the package name that spack resolved from all the ones that `provides('scalapack')`?

[as later this code is using `spec.dependencies_dict()` - and the correct package name appears to be required here]

Reply:
Ah - should have been obvious.

`spec['scalapack'].name`